platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/MerkleBenchmarkUtils.java

```diff
@@ -137,6 +138,13 @@ private static <T extends MerkleNode> T testSynchronization(
 
             final MerkleNode generatedTree = learner.getRoot();
             return (T) generatedTree;
+        } finally {
+            try {
+                streams.close();
+            } catch (IOException e) {
+                // test code, no danger
+                e.printStackTrace();
+            }
```

Comment:
Similar to a prior fix surrounding stream.close(), I'd really prefer the case of double-closing to be handled in the stream implementation, rather than being worked around everywhere we deal with streams.

Note that I still want exceptions unrelated to double-closing to be revealed in tests.

Reply:
Fair enough. I changed the fix.